bigtable/bigtable.go

```diff
@@ -453,9 +461,53 @@ func prefixSuccessor(prefix string) string {
 	return string(ans)
 }
 
+// ReadIterationStats captures information about the iteration of rows or cells over the course of
+// a read, e.g. how many results were scanned in a read operation versus the results returned.
+type ReadIterationStats struct {
+	CellsReturnedCount int64
+	CellsSeenCount     int64
+	RowsReturnedCount  int64
+	RowsSeenCount      int64
+}
+
+// RequestLatencyStats provides a measurement of the latency of the request as it interacts with
+// different systems over its lifetime, e.g. how long the request took to execute within a frontend
+// server.
+type RequestLatencyStats struct {
+	FrontendServerLatency time.Duration
+}
+
+// FullReadStats captures all known information about a read.
+type FullReadStats struct {
+	ReadIterationStats  ReadIterationStats
+	RequestLatencyStats RequestLatencyStats
+}
+
+func makeFullReadStats(reqStats *btpb.RequestStats) FullReadStats {
```

Comment:
NIT: a small comment before this function will be useful for future Golang client maintainers.